bahmnicore-omod/src/main/java/org/bahmni/module/bahmnicore/web/v1_0/controller/BahmniPatientProfileResource.java

```diff
@@ -246,13 +246,23 @@ private String getValueFromMap(Map<String, Object> jsonMap, String key) {
         Object value = jsonMap.get(key);
         return ObjectUtils.toString(value);
     }
-
+    
     private Relationship createRelationship(Person currentPerson, Map<String, Object> relationshipJson) {
-        Relationship relationship = new Relationship(currentPerson,
-                getPerson((Map<String, Object>) relationshipJson.get("personB")),
-                getRelationshipType((Map<String, Object>) relationshipJson.get("relationshipType")));
+        Relationship relationship;
+        String direction = (String) relationshipJson.get("direction");
+        
+        if (direction.equals("aToB")) {
+            relationship = new Relationship(currentPerson,
+                    getPerson((Map<String, Object>) relationshipJson.get("person")),
```

Comment:
`getPerson()` and `getRelationshipType()` can be written only once, before the `if`.

Reply:
Could you implement a line where we make it backward compatible?
Something that would check if `relationshipJson.get("person"))` is null, and if so try to get `relationshipJson.get("personB"))`. If the latter is not null, we set the direction to be `A_to_B`.